### PR TITLE
GF coldpool updates and bugfix for radiation coupling

### PR DESCRIPTION
**Pull request by haiqinli**
Enter a description of this PR. This should include why this PR was created, and what it does.
-- A bugfix to add GF clouds on top of the clouds provided by the other options. The GF coldpool parameterization impacts on momentum is also updated. 

Testing and relations to other Pull Requests should be added as subsequent comments.
-- This PR has been tested on Jet.

### CODE REVIEW

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (barlage) [inline src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F:43]: clearer to put 2 on its own line

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (barlage) [inline src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F]: remove commented line

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (barlage) [inline src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F]: indent for this if block

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (barlage) [inline src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F]: indent for this if block

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (barlage) [inline src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F]: indenting is messed up here

File: src/core_atmosphere/physics/mpas_atmphys_driver_cloudiness.F
Comment (barlage) [inline src/core_atmosphere/physics/mpas_atmphys_driver_cloudiness.F]: for future merge purposes, I would avoid removing lines in the mpas core code

File: src/core_atmosphere/physics/mpas_atmphys_driver_cloudiness.F
Comment (barlage) [inline src/core_atmosphere/physics/mpas_atmphys_driver_cloudiness.F:244]: is the intention to always overwrite the mynn cldfrac calculated above when cu_grell_freitas?

File: src/core_atmosphere/physics/mpas_atmphys_driver_cloudiness.F
Comment (barlage) [inline src/core_atmosphere/physics/mpas_atmphys_driver_cloudiness.F:351]: similar to above, is the intention to always overwrite the fields calculated in the case statement above?

**Review by barlage — COMMENTED** [review]
A few formatting suggestions (take em or leave em). And a question about the intention of taking the two blocks out of the case statement

**Comment by barlage** [comment]
@haiqinli maybe I understand now that these are additive on top of the other schemes.

File: src/core_atmosphere/physics/mpas_atmphys_driver_cloudiness.F
Comment (barlage) [reply src/core_atmosphere/physics/mpas_atmphys_driver_cloudiness.F:351]: I looked at cal_gfcld and understand now what is happening.

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (haiqinli) [reply src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F]: The if block is used to only apply the convective clouds between 300hPa and 600hPa above surface.

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (haiqinli) [reply src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F]: The sum of cloud depth (dpsum) should be large then 0 (dpsum is initialized as 0).

**Comment by haiqinli** [comment]
@barlage Yes, the convective clouds are additive on top of the other schemes. Joe pointed out this issue yesterday. I will also update the formatting following your suggestions. Thanks!

**Review by joeolson42 — APPROVED** [review]
Approved after the minor formatting changes called out by Mike.

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (barlage) [reply src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F]: my suggestion is to indent these lines in the if block

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (barlage) [reply src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F]: same, just a suggestion to indent the lines in the if block

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (haiqinli) [reply src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F:43]: Updated following your suggestions. Thanks.

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (haiqinli) [reply src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F]: Removed following your suggestions. Thanks.

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (haiqinli) [reply src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F]: Updated following your suggestions. Thanks.

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (haiqinli) [reply src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F]: Updated following your suggestions. Thanks.

File: src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F
Comment (haiqinli) [reply src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F]: Updated following your suggestions. Thanks.

**Comment by jderrico-noaa** [comment]
@barlage do the changes sufficiently address the formatting concerns or should I hold off on merging?

**Review by tanyasmirnova — APPROVED** [review]

**Review by barlage — APPROVED** [review]

**Pull request merged**
